### PR TITLE
Adding CFL output and test

### PR DESCRIPTION
**Pull request by mpentek**

### CODE REVIEW

**Review by AndreasWinterstein — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
Guys, I've some comments on this PR. Next time before approving and merging please wait for the approval of at least one of the official maintainers of the application you are merging to, specially if this application is a candidate to become a Core application (as the FluidDynamicsApplication is).

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:21]: I think this description is not correct (copy & paste error?)

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:29]: Same

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: I understand that this process is just for outputting the local CFL. Then, which is the intention of this `cfl_threshold`. Why do we want to limit it if we just want to output it?

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:68]: If you save the parameters as member variable you don't need these to be a member variable too.

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:68]: > If you save the parameters as member variable you don't need these (`interval`, `cfl_threshold`, `format`, `output_step` and so on) to be a member variable too. If you want to have them as member variables to avoid accessing the parameters container each time, I suggest to do it in the constructor after the validation of the defaults.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:153]: According to the standard, "private" methods in Python should be preceded by an `_` and written in `snake_case`.

Comment (philbucher) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:153]: Actually it is correct -> @rubenzorrilla check the style guide, we agreed on 
Public: PublicMethod
Protected: _ProtectedMethod
Private: __PrivateMethod
A long time ago
It is not yet done like that everywhere, but at least for new code it is suggested to follow this

**Review by rubenzorrilla — COMMENTED** [review]
Aside of the minor comments (which can be easily addressed in another PR), this makes sense for me.

Should we add this feature to the interface as we did with the drag?

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:153]: OK then. My apologies. @mpentek forget about my comment.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: The threshold value is not limiting, it just helps for an additional statistics value in the summarize. Many times you would not only be interested in how many elements have CFL above 1.0, but lets say 2.5.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:21]: Done

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:29]: Done

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:68]: Well, as you saw from the description, this takes as an example script the compute_drag_process which also deals with these values the same way in ExecuteInitialize().

Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:68]: Also the same holds for private function naming, that is why it is _EvaluateCFL() instead of _evaluate_cfl(), which would be the more canonic pythonic way.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: Understood. I'd rather call it `cfl_output_lower_bound` or something similar. I find that`cfl_threshold` is a bit misleading.

**Review by mpentek — COMMENTED** [review]
Adding it to the interface as drag would be very useful.

**Comment by mpentek** [comment]
> Guys, I've some comments on this PR. Next time before approving and merging please wait for the approval of at least one of the official maintainers of the application you are merging to, specially if this application is a candidate to become a Core application (as the FluidDynamicsApplication is).

Hi, yes, you are right. I restored the branch for some minor corrections.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:68]: > Well, as you saw from the description, this takes as an example script the compute_drag_process which also deals with these values the same way in ExecuteInitialize().

We did the drag process long time ago so don't take it as reference... :laughing: 
My point is that it is redundant to save both the parameters and these variables.

**Comment by rubenzorrilla** [comment]
> > Guys, I've some comments on this PR. Next time before approving and merging please wait for the approval of at least one of the official maintainers of the application you are merging to, specially if this application is a candidate to become a Core application (as the FluidDynamicsApplication is).
> 
> Hi, yes, you are right. I restored the branch for some minor corrections.

Cool. Thanks!

**Comment by mpentek** [comment]
@rubenzorrilla the new PR is #5988, for whatever reason I could not restore this one correctly.

**Comment by rubenzorrilla** [comment]
> Adding it to the interface as drag would be very useful.

We can add it after the release. Let me open an issue as a reminder.

File: applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py
Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: It also does not bound anything, it is just a threshold or limit value (would that be better) for the _Summarize function.

Comment (mpentek) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: or custom limit?

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py:38]: > or custom limit?
I'd go for either `cfl_output_limit` or `cfl_output_custom_limit` (the one you like the most). What should be clear is that we are not limiting the CFL, just the output information.

**Pull request merged**
